sdk/python/kfp/compiler/compiler_test.py

```diff
@@ -1490,5 +1490,161 @@ def pipeline_with_input(boolean: bool = False):
             .default_value.bool_value, True)
 
 
+class TestYamlComments(unittest.TestCase):
+
+    def test_comments_include_inputs_and_outputs_and_pipeline_name(self):
+
+        @dsl.component
+        def identity(string: str, model: bool) -> str:
+            return string
+
+        @dsl.pipeline()
+        def my_pipeline(sample_input1: bool = True,
+                        sample_input2: str = 'string') -> str:
+            op1 = identity(string=sample_input2, model=sample_input1)
+            result = op1.output
+            return result
+
+        with tempfile.TemporaryDirectory() as tmpdir:
+            pipeline_spec_path = os.path.join(tmpdir, 'output.yaml')
+            compiler.Compiler().compile(
+                pipeline_func=my_pipeline, package_path=pipeline_spec_path)
+
+            inputs_string = '# Inputs: \n#    sample_input1: bool [Default: True]\n#    sample_input2: str [Default: string]'
+            outputs_string = '# Outputs: \n#    Output: str'
+            name_string = '# Name: my-pipeline'
+
+            # test name is in comments
+            with open(pipeline_spec_path, 'r+') as f:
+                yaml_content = f.read()
```

Comment:
nit: These tests can be a bit more readable by re-reading in `yaml_content` once within the `tempfile` context manager, then letting the context manager exit, with the assertions in the method's outermost scope. This makes it clear that the code that follows don't depend on the temporary directory anymore.